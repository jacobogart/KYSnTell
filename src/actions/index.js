export const setLocations = (locations) => ({
  type: 'SET_LOCATIONS',
  locations
})