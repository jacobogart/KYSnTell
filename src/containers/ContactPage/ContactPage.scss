.contact-form {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 60vh;
  margin-top: 80px;
}

.contact-button,
.contact-input {
  background-color: white;
  font-size: 30px;
  padding: 8px;
}

.contact-list {
  background-color: grey;
  list-style: none;
}

.contact-form {
  display: flex;
  flex-direction: column;
}