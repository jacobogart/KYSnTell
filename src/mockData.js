import { locationCleaner } from "./api/cleaners";

export const origin = {
  lat: 38.9119463, 
  lng: -94.4136213 
}

export const destination = { 
  lat: "39.1952922", 
  lng: "-94.6817269" 
}

export const distance = '18.3 mi';

export const distanceResult = {
  rows: [
    {
      elements: [
        {
          distance: {
            text: distance
          }
        }
      ]
    }
  ]
}


export const fetchDistanceURL = "https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/distancematrix/json?origins=38.9119463,-94.4136213&destinations=39.1952922,-94.6817269&units=imperial&key=AIzaSyDHKnndIp7I8oymrxO0sDwpQUdZvfN1sqc"

export const latLongResult = {
  results: [
    {
      geometry: {
        location: origin
      }
    }
  ]
}


export const fetchLatLongURL = "https://maps.googleapis.com/maps/api/geocode/json?address=64081&key=AIzaSyDHKnndIp7I8oymrxO0sDwpQUdZvfN1sqc"

export const locations = [
  {
    title: "Neighborhood Health",
    link: "https://gettested.cdc.gov/gettested_redirect/21952",
    id: "tag:hivtest.org,2000-03-20:/search/Detailx.asp?id=21952",
    streetAddress: "617 S 8th St",
    locality: "Nashville",
    region: "TN",
    postalCode: "37206-3819",
    telephone: "(615)-227-3000",
    categories: "",
    point: {
      lat: "36.1678930",
      long: "-86.7547047"
    },
    updated: "2019-02-08T17:19:32.0000000-05:00"
  },
  {
    title: "Street Works",
    link: "https://gettested.cdc.gov/gettested_redirect/110792",
    id: "tag:hivtest.org,2008-05-21:/search/Detailx.asp?id=110792",
    streetAddress: "520 Sylvan St",
    locality: "Nashville",
    region: "TN",
    postalCode: "37206",
    telephone: "(615)-259-7676",
    categories: "",
    point: {
      lat: "36.1665114",
      long: "-86.7605114"
    },
    updated: "2018-08-02T18:05:30.0000000-04:00"
  }
];

export const locationsResult = {
  services: [
    {
      providers: locations
    }
  ]
}

export const cleanLocations = locations.map(location => locationCleaner(location))

export const fetchLocationsURL = "https://cors-anywhere.herokuapp.com/https://locator.hiv.gov/data?lat=38.9119463&long=-94.4136213&distance=10"

export const contact = '+18168682142';

export const contacts = [contact, contact]

export const message = 'Test message.'

export const sendMessageOptions = {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({ to: contact, message })
}

export const resolvedMessages = [{ ok: true }, { ok: true }];

export const cleanLocation = {
  title: 'Neighborhood Health',
  link: 'https://gettested.cdc.gov/gettested_redirect/21952',
  id: '21952',
  address: '617 S 8th St;Nashville, TN 37206-3819',
  telephone: '(615)-227-3000',
  point: { lat: '36.1678930', lng: '-86.7547047' }
}

export const details = {
  additionalNotes: "Test",
  diagnosis: "Herpes",
  timeFrame: "year"
}

export const user = {
  location: origin
}

export const searchChangeEvent = {
  target: {
    name: 'zipcode',
    value: '64081'
  }
}

export const submitEvent = {
  preventDefault: jest.fn()
}

export const history = {
  push: jest.fn()
}

export const setUserLocation = jest.fn();

export const setLocations = jest.fn();

export const setContacts = jest.fn();

export const fullMessage = 'Hello, we have been informed by an anonymous sexual partner that you may have been exposed to Herpes in the last year. While this is no cause for alarm, we do recommend getting tested at your earliest convenience. To find a testing center near you, please visit kysntell.com, or contact your preferred healthcare provider. Additional notes from partner: Test'

export const messageWONotes = "Hello, we have been informed by an anonymous sexual partner that you may have been exposed to Herpes in the last year. While this is no cause for alarm, we do recommend getting tested at your earliest convenience. To find a testing center near you, please visit kysntell.com, or contact your preferred healthcare provider."

export const previewEvent = {
  target: {
    blur: jest.fn()
  }
}

export const messageContext = {
  focus: jest.fn(),
  blur: jest.fn(),
  innerText: 'Test'
}

 export const rejectedMessages = [{ ok: false, contact }]

 export const routerLocation = {
  pathname:'/kys/locations/21952'
 }

export const detailsChangeEvent = {
  target: {
    name: 'diagnosis',
    value: 'Herpes'
  }
}

export const setDetails = jest.fn();

export const contactChangeEvent = {
  target: {
    value: contact
  }
}

export const contactBadChangeEvent = {
  target: {
    value: 'test'
  }
}

export const searchDefaultState = {
  zipcode: '',
  distance: '',
  isLoading: false,
  error: false
}