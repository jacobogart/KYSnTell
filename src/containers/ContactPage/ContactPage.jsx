import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setContacts } from '../../actions';
import phone from 'phone';
class ContactPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      contacts: [],
      input: '',
      error: false
    };
  }
  
  handleChange = (e) => {
    this.setState({ input: e.target.value, error: false });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const formattedContact = phone(this.state.input);
    formattedContact.length 
      ? this.setState({ contacts: [...this.state.contacts, formattedContact[0]], input: '' })
      : this.setState({ error: true, input: '' })
  }

  storeContacts = () => {
    this.props.setContacts(this.state.contacts)
    this.props.history.push('/tell/details')
  }

  render() {
    const contactList = this.state.contacts.map(contact => {
      let numbers = contact.split('');
      const [ , , a, b, c, d, e, f, g, h, i, j] = numbers;
      return <li>{`(${a}${b}${c}) ${d}${e}${f}-${g}${h}${i}${j}`}</li>})
    return (
      <div className="ContactPage">
        <div className="contact-list-container">
          <ol className="contact-list">
            {contactList}
          </ol>
        </div>
        <form 
          onSubmit={this.handleSubmit} 
          className="contact-form"
        >
          <input 
            type="text" 
            className="contact-input" 
            placeholder='(---) --- ----' 
            value={this.state.input}
            onChange={this.handleChange}
          />
          {this.state.error &&
          <p className="error">Please enter a valid 10-digit phone number please.</p>}
          <button type="submit" className="contact-button">
            Add Contact
          </button>
          <button className="contact-button" onClick={this.storeContacts}>
            Next Page
        </button>
        </form>  
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setContacts: (contacts) => dispatch(setContacts(contacts))
})

export default connect(null, mapDispatchToProps)(ContactPage);