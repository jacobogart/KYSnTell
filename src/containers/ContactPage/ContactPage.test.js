import React from 'react';
import { shallow } from 'enzyme';
import { ContactPage, mapDispatchToProps } from './ContactPage';
import * as mock from '../../mockData'
import { setContacts } from '../../actions';

describe('ContactPage', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(
      <ContactPage 
        setContacts={mock.setContacts}
        history={mock.history}
      />);
    instance = wrapper.instance();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should match snapshot if error is true', () => {
    instance.setState({ error: true })
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    const defaultState = { contacts: [], input: '', error: false }
    expect(wrapper.state()).toEqual(defaultState);
  });

  describe('Event Listeners', () => {
    it('should invoke handleSubmit when the form submits', () => {
      wrapper.find('.contact-form').simulate('submit', mock.submitEvent);
      expect(mock.submitEvent.preventDefault).toHaveBeenCalled();
    });

    it('should invoke handleChange when the input is changed', () => {
      wrapper.find('.contact-input').simulate('change', mock.contactChangeEvent);
      expect(wrapper.state('input')).toEqual(mock.contact);
    })
  });

  describe('handleChange', () => {
    it('should set state to the input value', () => {
      instance.handleChange(mock.contactChangeEvent);
      expect(wrapper.state('input')).toEqual(mock.contact);
    });

    it('should set the error property to false', () => {
      instance.setState({ error: true });
      instance.handleChange(mock.contactChangeEvent);
      expect(wrapper.state('error')).toEqual(false);
    });
  });

  describe('handleSubmit', () => {
    it('should add the contact to state if it is correct', () => {
      expect(wrapper.state('contacts')).toEqual([]);
      instance.handleChange(mock.contactChangeEvent);
      instance.handleSubmit(mock.submitEvent);
      expect(wrapper.state('contacts')).toEqual([mock.contact]);
    });

    it('should set the error state to true is contact is not correct', () => {
      expect(wrapper.state('error')).toEqual(false);
      instance.handleChange(mock.contactBadChangeEvent);
      expect(wrapper.state('error')).toEqual(false);
    });

    it('should reset the input value either way', () => {
      instance.handleChange(mock.contactChangeEvent);
      expect(wrapper.state('input')).toEqual(mock.contact);
      instance.handleSubmit(mock.submitEvent);
      expect(wrapper.state('input')).toEqual('');

      instance.handleChange(mock.contactBadChangeEvent);
      expect(wrapper.state('input')).toEqual('test');
      instance.handleSubmit(mock.submitEvent);
      expect(wrapper.state('input')).toEqual('');
    });
  });

  describe('storeContacts', () => {
    it('should invoke setContacts', () => {
      instance.handleChange(mock.contactChangeEvent);
      instance.handleSubmit(mock.submitEvent);
      instance.storeContacts();
      expect(mock.setContacts).toHaveBeenCalledWith([mock.contact]);
    });

    it('should push the router to the correct route', () => {
      instance.storeContacts();
      expect(mock.history.push).toHaveBeenCalledWith('/tell/details')
    });
  });

  describe('mockDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    const actionToDispatch = setContacts(mock.contacts);
    mappedProps.setContacts(mock.contacts);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});