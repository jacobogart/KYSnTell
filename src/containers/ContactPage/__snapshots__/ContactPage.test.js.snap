// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage should match snapshot 1`] = `
<div
  className="ContactPage"
>
  <div
    className="contact-list-container"
  >
    <ol
      className="contact-list"
    />
  </div>
  <form
    className="contact-form"
    onSubmit={[Function]}
  >
    <input
      className="contact-input"
      onChange={[Function]}
      placeholder="(---) --- ----"
      type="text"
      value=""
    />
    <button
      className="contact-button"
      type="submit"
    >
      Add Contact
    </button>
    <button
      className="contact-button"
      onClick={[Function]}
    >
      Next Page
    </button>
  </form>
</div>
`;

exports[`ContactPage should match snapshot if error is true 1`] = `
<div
  className="ContactPage"
>
  <div
    className="contact-list-container"
  >
    <ol
      className="contact-list"
    />
  </div>
  <form
    className="contact-form"
    onSubmit={[Function]}
  >
    <input
      className="contact-input"
      onChange={[Function]}
      placeholder="(---) --- ----"
      type="text"
      value=""
    />
    <p
      className="error"
    >
      Please enter a valid 10-digit phone number please.
    </p>
    <button
      className="contact-button"
      type="submit"
    >
      Add Contact
    </button>
    <button
      className="contact-button"
      onClick={[Function]}
    >
      Next Page
    </button>
  </form>
</div>
`;
